db/write_controller.cc

```diff
@@ -20,10 +21,11 @@ std::unique_ptr<WriteControllerToken> WriteController::GetStopToken() {
 
 std::unique_ptr<WriteControllerToken> WriteController::GetDelayToken(
     uint64_t write_rate) {
-  total_delayed_++;
-  // Reset counters.
-  last_refill_time_ = 0;
-  bytes_left_ = 0;
+  if (0 == total_delayed_++) {
```

Comment:
Not an expert at this code, and I am also trying to reverse-engineering what it does.
One question: after a new super version is installed, we may call `SetupDelay()` in which a delay token is acquired via `GetDelayToken()` and the internal `total_delayed_` will become 1. Later, the first writer comes in. 
It will still see `NeedsDelay()` as true, and will enter `DelayWrite()`. Inside `DelayWrite()`, when it reaches
```
while (write_controller_.NeedsDelay()) {
...
// sleep for 1ms
}
```
It seems it will still sleep.
Maybe I am missing something...

Reply:
That should never be reached if GetDelay returns 0, which we try to do. See e.g. the `StartFilled` test. :)

Reply:
My bad. I missed the `if (delay > 0)` part just before that.